flask_multipass_cern.py

```diff
@@ -98,10 +114,18 @@ def get_members(self):
     def has_member(self, identifier):
         cache_key = f'flask-multipass-cern:{self.provider.name}:groups:{identifier}'
         all_groups = self.provider.cache and self.provider.cache.get(cache_key)
+        should_refresh = self.provider.cache and self.provider.cache.get(f'{cache_key}:timestamp') is None
+        if all_groups is None or should_refresh:
+            try:
+                all_groups = {g.name.lower() for g in self.provider.get_identity_groups(identifier)}
+                if self.provider.cache:
+                    self.provider.cache.set(cache_key, all_groups, timeout=CACHE_LONG_TTL)
+                    self.provider.cache.set(f'{cache_key}:timestamp', datetime.now(), timeout=CACHE_TTL)
+            except HTTPError as err:
+                _logger.exception(err)
```

Comment:
If we don't have group information and the request failed I'd log this differently (because that means a user will most likely be refused access due to the error) and explicitly return `False` (the code below would probably fail otherwise).